docs/sql-performance-tuning.md

```diff
@@ -228,6 +228,8 @@ The "REPARTITION_BY_RANGE" hint must have column names and a partition number is
     SELECT /*+ REPARTITION */ * FROM t
     SELECT /*+ REPARTITION_BY_RANGE(c) */ * FROM t
     SELECT /*+ REPARTITION_BY_RANGE(3, c) */ * FROM t
+    SELECT /*+ ADAPTIVE_REPARTITION */ * FROM t
+    SELECT /*+ ADAPTIVE_REPARTITION(c) */ * FROM t
```

Comment:
I'm not good at naming, let's get more opinions. cc @dongjoon-hyun @HyukjinKwon @maropu @viirya 

Basically this hint means the user wants to add a shuffle right before data writing, to redistribute data and avoid small files/skewed partitions. It's different from normal repartition as this allows more aggressive optimization from AQE.

Reply:
`REPARTITION_BY_AQE`?

Reply:
+1 for `REPARTITION_BY_AQE `